plugins/batterystat.koplugin/main.lua

```diff
@@ -207,10 +202,47 @@ function BatteryStat:showStatistics()
     table.insert(kv_pairs, "----------")
     table.insert(kv_pairs, {_("Historical records are dumped to"), ""})
     table.insert(kv_pairs, {self.dump_file, ""})
-    UIManager:show(KeyValuePage:new{
+    table.insert(kv_pairs, "----------")
+    table.insert(kv_pairs, {_("Should you like to reset the data,"), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    table.insert(kv_pairs, {_("please tap here."), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    self.kv_page = KeyValuePage:new{
         title = _("Battery statistics"),
         kv_pairs = kv_pairs,
-    })
+    }
+    UIManager:show(self.kv_page)
+end
+
+function BatteryStat:reset(withCharging, withDischarging)
+    self:dumpToText()
+    self.awake = Usage:new()
+    self.sleeping = Usage:new()
+
+    if withCharging then
+        self.charging = Usage:new()
+    end
+    if withDischarging then
+        self.discharging = Usage:new()
+    end
+end
+
+function BatteryStat:resetAll()
+    self:reset(true, true)
+    self.charging_state = State:new()
+    self.awake_state = State:new()
+end
+
+function BatteryStat:restart()
+    assert(self.kv_page ~= nil)
```

Comment:
Yes, but I found it a little bit unnatural. Comparing the following two implementations,
```lua
function A:f(p1, p2)
end
dbg:guard(A, "f", function(self, p1, p2)
  assert(p1 > 0)
end)
```
```lua
function A:f(p1, p2)
  dbg:dassert(p1 > 0)
end
```
The second one seems much cleaner, and guard() can only check input parameters, but dassert() can do almost everything.